lib/customisation.nix

```diff
@@ -269,9 +269,8 @@ rec {
     let self = f self // {
           newScope = scope: newScope (self // scope);
           callPackage = self.newScope {};
-          overrideScope = g: lib.warn
-            "`overrideScope` (from `lib.makeScope`) is deprecated. Do `overrideScope' (self: super: { … })` instead of `overrideScope (super: self: { … })`. All other overrides have the parameters in that order, including other definitions of `overrideScope`. This was the only definition violating the pattern."
-            (makeScope newScope (lib.fixedPoints.extends (lib.flip g) f));
+          overrideScope = g: makeScope newScope (lib.fixedPoints.extends g f);
+          # Add a warning after 23.11 and remove after 24.05.
```

Comment:
I think the timing should be much more relaxed. Somebody could be using `overrideScope` while still using 23.05 and get the warning that it's deprecated and `overrideScope'` should be used, but then they upgrade to 24.05 on the same day and suddenly `.overrideScope'` is gone, even though they only upgraded by 2 release.

Reply:
I meant remove it after 24.05 is branched off

Reply:
I'll tweak these, push coming soon.

Reply:
Agree.  Looks like this was resolved in the latest push.

Reply:
Not really: https://github.com/NixOS/nixpkgs/pull/248988#discussion_r1293613666